themes/bootstrap3/templates/RecordTab/holdingsils.phtml

```diff
@@ -10,7 +10,8 @@
     $urls = $this->record($this->driver)->getLinkDetails($openUrlActive);
     $offlineMode = $this->ils()->getOfflineMode();
     try {
-        $holdings = $this->driver->getRealTimeHoldings();
+        $options = $this->tab->getPaginatorOptions();
+        $holdings = $this->driver->getRealTimeHoldings($options);
```

Comment:
Maybe this is more complicated than I'm thinking (I confess I haven't dug into the code in a lot of depth yet), but my hope is that the output of getRealTimeHoldings could include both the count and the current page, so that we could then pass the data needed by getPaginatorOptions from there without having to either cache values inside the catalog connection or repeat lookups. If that's feasible, I think it would make the code a little more predictable and less potentially vulnerable to side effects. Again, if this is not straightforward, I'm willing to try to put some time into adapting your work into a new implementation -- just let me know if you need me to do so.

Reply:
I added the count to the output of `getRealTimeHoldings`. But I see a problem in getting the current page of the paginator there. I can't access the request object for getting the GET parameter `page`. Would you have any suggestions concerning that? At the moment, getting the page is done in `HoldingsILS->getCurrentPage`.

Reply:
The easiest solution may be to inject the request object into the HoldingsILS service by modifying the factory (https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/RecordTab/HoldingsILSFactory.php) and the HoldingsILS constructor. I believe the service is simply named 'request' inside the container. Let me know if you have trouble with this!

Reply:
The thing is that I already get the 'page' parameter in HoldingsILS (method `getCurrentPage()`). Getting the request object there is already working. But as you suggested in your first comment above, the 'page' parameter should be in the output of `getRealTimeHoldings`. But this method is in `IlsAwareTrait` where I can't access the request object. I pushed my changes now so you can take a look at it.

Reply:
I see. But all IlsAwareTrait::getRealTimeHoldings is doing is delegating to $this->holdLogic->getHoldings(), which is defined in the VuFind\ILS\Logic\Holdings class. What I meant to suggest was injecting the request into VuFind\ILS\Logic\Holdings and adjusting the work there... but for some reason I accidentally pasted the wrong class name and factory into my previous comment. Sorry about the confusion! Right now the Holds and TitleHolds logic classes share this factory: https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/ILS/Logic/LogicFactory.php -- it might be necessary to either conditionally inject the request or split the factory to make things work right. Is that any help, or do you think we're still on the wrong track?